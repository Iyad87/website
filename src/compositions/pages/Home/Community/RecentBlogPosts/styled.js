import styled from 'styled-components';
import { textLight } from 'styles/colours';
import { spacing } from 'styles/spacing';
import breakpoints from 'styles/breakpoints';

export const BlogPosts = styled.ul`
  display: flex;
  flex-wrap: wrap;
  margin: ${spacing(-1.25)};
  margin-bottom: ${spacing(6.25)};
`;

export const BlogPost = styled.li`
  display: block;
  padding: ${spacing(1.25)};
  width: 100%;

  ${breakpoints.tablet`
    width: 50%;
  `};
`;

export const Title = styled.h3`
  color: ${textLight};
  font-weight: 500;
  font-size: 2.25rem;
  margin-bottom: ${spacing(5)};
`;
