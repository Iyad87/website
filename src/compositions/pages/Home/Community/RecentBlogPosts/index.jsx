import React from 'react';
import Post from '../../../../../components/Post';
import { BlogPost, BlogPosts, Title } from './styled';

import Button from '../../../../../components/Button';
import Link from '../../../../../components/Link';

const RecentBlogPosts = () => (
  <section>
    <header>
      <Title>Recent Blog Posts</Title>
    </header>
    <BlogPosts>
      <BlogPost>
        <Post
          date="2018-05-18T14:38:23.996+01:00"
          image="https://placekitten.com/128/128"
          sourceLink="/people/nuno-job"
          sourceText="Nuno Job"
          title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
        >
          I think its one of the simplest yet most powerful extensions inVision
          hasn&apos;t launched. As animation and transition in inVision evolve
          it should grow even more powerful
        </Post>
      </BlogPost>
      <BlogPost>
        <Post
          date="2018-05-18T14:38:23.996+01:00"
          image="https://placekitten.com/128/128"
          sourceLink="/people/nuno-job"
          sourceText="Nuno Job"
          title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
        >
          I think its one of the simplest yet most powerful extensions inVision
          hasn&apos;t launched. As animation and transition in inVision evolve
          it should grow even more powerful
        </Post>
      </BlogPost>
    </BlogPosts>
    <Button component={Link} href="/blog">
      View Blog
    </Button>
  </section>
);

export default RecentBlogPosts;
