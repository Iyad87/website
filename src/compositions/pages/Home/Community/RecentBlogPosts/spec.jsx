import React from 'react';
import { shallow } from 'enzyme';
import RecentBlogPosts from './index';

test('compositions/Home/Community/RecentBlogPosts snapshot', () => {
  const wrapper = shallow(<RecentBlogPosts />);

  expect(wrapper).toMatchSnapshot();
});
