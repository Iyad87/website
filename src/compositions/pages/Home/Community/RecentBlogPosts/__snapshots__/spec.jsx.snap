// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compositions/Home/Community/RecentBlogPosts snapshot 1`] = `
<section>
  <header>
    <styled.h3>
      Recent Blog Posts
    </styled.h3>
  </header>
  <styled.ul>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Nuno Job"
        stretch={true}
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched. As animation and transition in inVision evolve it should grow even more powerful
      </Post>
    </styled.li>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Nuno Job"
        stretch={true}
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched. As animation and transition in inVision evolve it should grow even more powerful
      </Post>
    </styled.li>
  </styled.ul>
  <Button
    component={[Function]}
    href="/blog"
  >
    View Blog
  </Button>
</section>
`;
